internal/backend/db_mailbox.go

```diff
@@ -211,15 +211,15 @@ func DBGetMailboxByID(ctx context.Context, client *ent.Client, id imap.InternalM
 	return client.Mailbox.Query().Where(mailbox.MailboxID(id)).Only(ctx)
 }
 
-func DBGetMailboxMessages(ctx context.Context, mbox *ent.Mailbox) ([]*ent.UID, error) {
+func DBGetMailboxMessages(ctx context.Context, client *ent.Client, mbox *ent.Mailbox) ([]*ent.UID, error) {
 	return mbox.QueryUIDs().WithMessage().All(ctx)
 }
 
 func DBGetMailboxRecentCount(ctx context.Context, mbox *ent.Mailbox) (int, error) {
 	return mbox.QueryUIDs().Where(uid.Recent(true)).Count(ctx)
 }
 
-func DBGetMailboxMessagesForNewSnapshot(ctx context.Context, mbox *ent.Mailbox) ([]*ent.UID, error) {
+func DBGetMailboxMessagesForNewSnapshot(ctx context.Context, client *ent.Client, mbox *ent.Mailbox) ([]*ent.UID, error) {
```

Comment:
or here?

Reply:
While we don't need to use the client object, it is there to enforce the DB type use and to make sure we have acquire the lock. Otherwise, these queries will run at any moment, bypassing the lock and causing the infamous "database locked" error.